hathor/indexes/tips_index.py

```diff
@@ -61,3 +65,27 @@ def update_tx(self, tx: BaseTransaction, *, relax_assert: bool = False) -> None:
     @abstractmethod
     def __getitem__(self, index: float) -> Set[Interval]:
         raise NotImplementedError
+
+    def iter_mempool_tips_like(self, tx_storage: 'TransactionStorage') -> Iterator[Transaction]:
+        for interval in self[tx_storage.latest_timestamp + 1]:
+            tx = tx_storage.get_transaction(interval.data)
+            tx_meta = tx.get_metadata()
+            # XXX: skip blocks and tx-tips that have already been confirmed
```

Comment:
Why would it happen?

Reply:
Blocks could appear if we're using the wrong index, and transactions on the tips index can normally have a `first_block`, but not on the mempool, so those are filtered-out.

Reply:
I fixed this problem by moving the implementation to the TransactionStorage, and now I assert that each tx is a Transaction, but still have to check for first_block.

Reply:
Great! Thanks!